shell/src/main/java/alluxio/cli/fs/command/job/JobAttempt.java

```diff
@@ -43,13 +43,16 @@ protected JobAttempt(JobMasterClient client, RetryPolicy retryPolicy) {
    * @return true if an attempt was made, false if attempts ran out
    */
   public boolean run() {
-    if (mRetryPolicy.attempt()) {
+    while (mRetryPolicy.attempt()) {
       mJobId = null;
       try {
         mJobId = mClient.run(getJobConfig());
       } catch (IOException e) {
-        LOG.warn("Failed to start job", e);
-        System.out.println(String.format("Failed to start job with error: %s", e.getMessage()));
+        int retryCount = mRetryPolicy.getAttemptCount();
+        System.out.println(String.format("(Retry %d) Failed to start job with error: %s",
+            retryCount, e.getMessage()));
+        LOG.warn("(Retry %d) Failed to get status for job (jobId={})", retryCount, mJobId, e);
```

Comment:
https://docs.alluxio.io/os/user/edge/en/contributor/Code-Conventions.html#logging-conventions

// Recommended
LOG.warn("Failed to do something: {}", e.toString());
// Recommended
LOG.warn("Failed to do something: {}", e);
// Not recommended: this will print out the stack trace
LOG.warn("Failed to do something", e);
// Not recommended: the exception class name is not included
LOG.warn("Failed to do something", e.getMessage());

Reply:
add a {} for the `e`?